src/app/api/volunteer/auth/password-reset/update-password/route.ts

```diff
@@ -0,0 +1,44 @@
+import { IPasswordReset } from "@/common_utils/types";
+import { updateUserPassword } from "@server/firebase/passwordReset";
+import {
+  deletePasswordResetFieldByEmail,
+  getPasswordByToken,
+} from "@server/mongodb/actions/PasswordReset";
+import APIWrapper from "@server/utils/APIWrapper";
+
+type RequestData = {
+  token: string;
+  password: string;
+};
+
+export const POST = APIWrapper({
+  config: {
+    requireToken: true,
```

Comment:
This route should not require a token since the user will not be signed in for password reset.